source/_posts/2018-01-12-pager-duty-and-confluence.markdown

```diff
@@ -0,0 +1,256 @@
+---
+layout: post
+title: "Pager Duty and Confluence"
+date: 2018-01-12 18:19:49 +0000
+comments: true
+author: "Bryan Kroger bryan.kroger@grandrounds.com"
+categories: automation
+---
+My first project here at Grand Rounds was to pick up one of my favorite topics of IT and operations 
+in general: on-call rotations!  Making sure we have accurate alarming and alerting never seems like the
+most glamorous work in the world, but it is often very important and highly impactful.
+
+To that end, as with most things, we want to start with a data-driven approach to identifying and resolving
+our problems.  This is a pattern that I've used several times over the years for a fun little hack which
+allows us to build reports based on PagerDuty incident data.
+
+This gives us two graphs:
+
+* Number of incidents per service.
+* Amount of time each service spends in incident ( on average ).
+
+We fully expect that this code is a little rough around the edges, and that's okay.  We're building things
+here that are going to give us ideas and insight into our bigger picture items.  Ideally we want to
+spend as little time as possible gold plating this code because we're not 100% sure this will be something
+that we intend to live forever.  Just like in the startup business world we want to create the Minimal
+Viable Product ( MVP ) which allows us to prove out a business case.
+
+In this case we will prove out the value of incident reporting data.  This is a perfect example of a product
+that is valuable today, but becomes less valuable over time because it allows us to refine a process.
```

Comment:
I might briefly explain what's happening in your code. From what I can tell, it looks like you're collating a range of PagerDuty events into an HTML string, then sending that string to JIRA to build a readable page there. Is that right?

It might also be helpful to explain the context for this specific implementation - there isn't a JIRA client written in Ruby that can build the expected HTML string for you, for example, or you're trying to render a custom format because you want a specific look for your page, etc.

Reply:
I've looked for clients over the years, and some do some parts, but not others.  The thing I keep coming back to is that these are just dead simple API calls, and all I'm really doing is making a fancy wrapper around a very simple idea.  I think what would be really helpful here is to show the actual results of the page that is created, however, that's actually not awesome or helpful in our case.  But it would sort of clear things up.